@xen-orchestra/web-core/lib/components/StackedBar.vue

```diff
@@ -0,0 +1,140 @@
+<!-- v1.0 -->
+<template>
+  <div class="stacked-bar">
+    <div class="wrapper">
+      <div
+        v-for="(segment, index) in computedSegments"
+        :key="index"
+        ref="segmentRefs"
+        v-tooltip="
+          shouldShowTooltip(index)
+            ? {
+                placement: 'top',
+                content: `${segment.percentage.toFixed(0)} %`,
+              }
+            : false
+        "
+        :style="{ width: segment.percentage + '%' }"
+        class="segment typo c4-semi-bold"
+        :class="segment.color"
+      >
+        <span class="segment-text">{{ segment.percentage.toFixed(0) }} %</span>
+      </div>
+      <div
+        v-if="availableSegmentValue > 0"
+        :style="{ width: availableSegmentValue + '%' }"
+        class="segment available typo c4-semi-bold"
+      ></div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, onMounted, onUnmounted, ref } from 'vue'
+import type { Color } from '@core/types/color.type'
+import { vTooltip } from '@core/directives/tooltip.directive'
+
+type StackedBarDataType = {
+  value: number
+  color: Color
+}
+
+const props = defineProps<{
+  segments: StackedBarDataType[]
+  maxValue: number
+}>()
+
+const segmentRefs = ref<HTMLElement[] | null>(null)
+const minWidth = 40
+
+const totalValue = computed(() => props.segments.reduce((acc, bar) => acc + bar.value, 0))
+
+const accurateMaxValue = computed(() => {
+  return props.maxValue > totalValue.value && props.maxValue !== 0 ? props.maxValue : totalValue.value
+})
```

Comment:
You can use the same logic as in the `DonutChart`:
```suggestion
const max = computed(() => Math.max(props.maxValue ?? 0, totalValue.value))
```